src/dashboard/events/cd-dashboard-upcoming-event.vue

```diff
@@ -0,0 +1,216 @@
+    directives: {
+      ImgFallback,
+    },
+    filters: {
+      cdDateFormatter,
+      cdTimeFormatter,
+    },
+    data() {
+      return {
+        orders: null,
+      };
+    },
+    computed: {
+      ...mapGetters(['loggedInUser']),
+      eventDate() {
+        return EventUtils.getNextStartTime(this.event);
+      },
+      remainingTickets() {
+        return this.totalNinjaTickets - this.bookedNinjaTickets;
+      },
+      bookedNinjaTickets() {
+        return this.event.sessions.reduce((acc1, session) =>
+          acc1 + session.tickets.filter(ticket => ticket.type === 'ninja').reduce((acc2, ticket) =>
+            acc2 + ticket.approvedApplications, 0), 0);
+      },
+      totalNinjaTickets() {
+        return this.event.sessions.reduce((acc1, session) =>
+          acc1 + session.tickets.filter(ticket => ticket.type === 'ninja').reduce((acc2, ticket) =>
+            acc2 + ticket.quantity, 0), 0);
+      },
+      bookedMentorTickets() {
+        return this.event.sessions.reduce((acc1, session) =>
+          acc1 + session.tickets.filter(ticket => ticket.type === 'mentor').reduce((acc2, ticket) =>
+            acc2 + ticket.approvedApplications, 0), 0);
+      },
+      totalMentorTickets() {
+        return this.event.sessions.reduce((acc1, session) =>
+          acc1 + session.tickets.filter(ticket => ticket.type === 'mentor').reduce((acc2, ticket) =>
+            acc2 + ticket.quantity, 0), 0);
+      },
+      dojoImageUrl() {
+        return `https://s3-eu-west-1.amazonaws.com/zen-dojo-images/${this.dojo.id}`;
+      },
+      dojoImageFallbackImage() {
+        /* eslint-disable global-require */
+        return require('../../assets/avatars/dojo-default-logo.png');
+        /* eslint-enable global-require */
+      },
+      hasOrder() {
+        return this.orders && this.orders.length > 0;
+      },
+      formattedStartTime() {
+        return this.$options.filters.cdTimeFormatter(this.event.dates[0].startTime);
+      },
+      formattedEndTime() {
+        return this.$options.filters.cdTimeFormatter(this.event.dates[0].endTime);
+      },
+      ninjaTickets() {
+        return this.orders.reduce((acc, order) =>
+          acc + order.applications.filter(application => application.ticketType === 'ninja').length, 0);
+      },
+      mentorTickets() {
+        return this.orders.reduce((acc, order) =>
+          acc + order.applications.filter(application => application.ticketType === 'mentor').length, 0);
+      },
+      isChampion() {
+        return this.event.usersDojos.filter(usersDojo => usersDojo.userTypes.indexOf('champion') !== -1).length > 0;
+      },
+    },
+    methods: {
+      async loadOrders() {
+        const res = await EventsService.v3.getOrder(this.loggedInUser.id, {
+          params: {
+            query: { eventId: this.event.id },
+          },
+        });
+        this.orders = res.body.results;
+        this.$emit('loaded');
+      },
+    },
+    watch: {
+      event() {
```

Comment:
Can `event` change? If not, using a v-if on the parent avoid the condition on `created` and the watcher

Reply:
Yeah, because of the `[{}, {}]` in the parent component, the event eventually loads, causing event to be updated, hence this watch...